src/main/kotlin/view/FilterPanel.kt

```diff
@@ -51,68 +51,116 @@ class FilterPanel : View() {
+                            padding = Insets(0.0, 10.0, 0.0, 10.0)
+
+                            listview(engine.transformations) {
+                                prefWidth = 400.0
+                                selectionModel.selectedIndexProperty().onChange {
+                                    engine.setCurrentIndex(it)
+                                }
+                                engine.updateListSelection =
+                                    { selectionModel.select(engine.currIndex) }
+                            }
+                        }
+                        hbox {
+                            alignment = Pos.CENTER
+                            buttonbar {
+                                padding = Insets(20.0, 10.0, 20.0, 10.0)
+                                button("Undo").setOnAction { fileController.undo() }
+                                button("Redo").setOnAction { fileController.redo() }
+                                button("Revert").setOnAction { fileController.revert() }
+                            }
+                        }
                     }
+                    orientation = Orientation.VERTICAL
+                    setDividerPosition(0, 0.4)
                 }
+            }
+        }
 
-                hbox {
-                    alignment = Pos.CENTER
-                    padding = Insets(0.0, 10.0, 0.0, 10.0)
+        tab("Steganography") {
+            vbox {
+                splitpane {
+                    prefHeight = 1000.0
+                    val tabpane = tabpane {
+                        tabClosingPolicy = TabPane.TabClosingPolicy.UNAVAILABLE
+                        side = Side.LEFT
+                        tab("Encode/Decode Image") {
+                            content = EncodeImageTab(fileController, engine, engineController)
+                        }
 
-                    listview(engine.transformations) {
-                        prefWidth = 400.0
-                        selectionModel.selectedIndexProperty().onChange {
-                            engine.setCurrentIndex(it)
+                        tab("Encode/Decode Text") {
+                            content = EncodeTextTab(fileController, engine, engineController)
                         }
-                        engine.updateListSelection =
-                            { selectionModel.select(engine.currIndex) }
                     }
-                }
-                hbox {
-                    alignment = Pos.CENTER
-                    buttonbar {
-                        padding = Insets(20.0, 10.0, 20.0, 10.0)
-                        button("Undo").setOnAction { fileController.undo() }
-                        button("Redo").setOnAction { fileController.redo() }
-                        button("Revert").setOnAction { fileController.revert() }
+
+                    vbox {
+                        alignment = Pos.CENTER
+                        label("Result/Target Image") {
```

Comment:
Is this image view only used when decode image but not in decode text?

Reply:
Yes. I can maybe conditionally render it